fe/fe-core/src/main/java/com/starrocks/scheduler/TaskBuilder.java

```diff
@@ -44,6 +52,66 @@ public static Task buildMvTask(MaterializedView materializedView, String dbName)
         return task;
     }
 
+    public static Task rebuildMvTask(MaterializedView materializedView, String dbName,
+                                     Map<String, String> previousTaskProperties) {
+        Task task = new Task(getMvTaskName(materializedView.getId()));
+        task.setSource(Constants.TaskSource.MV);
+        task.setDbName(dbName);
+        previousTaskProperties.put(PartitionBasedMaterializedViewRefreshProcessor.MV_ID,
+                String.valueOf(materializedView.getId()));
+        task.setProperties(previousTaskProperties);
+        task.setDefinition(
+                "insert overwrite " + materializedView.getName() + " " + materializedView.getViewDefineSql());
+        task.setExpireTime(0L);
+        return task;
+    }
+
+    public static void mappingTaskTypeAndSetRefreshInfo(Task task, RefreshSchemeDesc refreshSchemeDesc)
+            throws DdlException {
+        MaterializedView.RefreshType refreshType = refreshSchemeDesc.getType();
+        if (refreshType == MaterializedView.RefreshType.MANUAL) {
+            task.setType(Constants.TaskType.MANUAL);
+        } else if (refreshType == MaterializedView.RefreshType.ASYNC) {
+            if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
+                AsyncRefreshSchemeDesc asyncRefreshSchemeDesc = (AsyncRefreshSchemeDesc) refreshSchemeDesc;
+                IntervalLiteral intervalLiteral = asyncRefreshSchemeDesc.getIntervalLiteral();
+                if (intervalLiteral == null) {
+                    task.setType(Constants.TaskType.EVENT_TRIGGERED);
+                } else {
+                    final IntLiteral step = (IntLiteral) asyncRefreshSchemeDesc.getIntervalLiteral().getValue();
+                    long startTime = Utils.getLongFromDateTime(asyncRefreshSchemeDesc.getStartTime());
+                    TaskSchedule taskSchedule = new TaskSchedule(startTime, step.getLongValue(),
+                            TimeUtils.convertUnitIdentifierToTimeUnit(intervalLiteral.getUnitIdentifier().getDescription()));
+                    task.setSchedule(taskSchedule);
+                    task.setType(Constants.TaskType.PERIODICAL);
+                }
+            }
+        }
+    }
+
+    public static void mappingTaskTypeAndSetRefreshInfo(Task task, MaterializedView materializedView)
```

Comment:
this function can be removed and replaced by the former one.

Reply:
no MaterializedView.AsyncRefreshContext and RefreshSchemaDes is different one.